src/s2i/bin/run

```diff
@@ -48,8 +48,17 @@ APP_HOME=$(readlink -f "${APP_HOME:-.}")
 PYTHONPATH="$(pwd)${PYTHONPATH:+:$PYTHONPATH}"
 cd "$APP_HOME"
 
-app_script_check="${APP_SCRIPT-}"
-APP_SCRIPT="${APP_SCRIPT-app.sh}"
+if [ -z "$APP_SCRIPT" ] && [ -z "$APP_FILE" ] && [ -z "$APP_MODULE" ]; then
+  # Set default values for APP_SCRIPT and APP_FILE only when all three APP_
+  # variables are not defined by user. This prevents a situation when
+  # APP_MODULE is defined to app:application but the app.py file is found as the
+  # APP_FILE and then executed by Python instead of gunicorn.
+  app_script_check="${APP_SCRIPT-}"
+  APP_SCRIPT="${APP_SCRIPT-app.sh}"
+  app_file_check="${APP_FILE-}"
+  APP_FILE="${APP_FILE-app.py}"
```

Comment:
We already know the variables are empty, so no need to check them again

```suggestion
  app_script_check=
  APP_SCRIPT="app.sh"
  app_file_check=
  APP_FILE="app.py"
```

Reply:
Thank you!

Reply:
Sources will need to be regenerated as well to get the changes to versioned files